# Daffa Asyqar Ahmad Khalisheka

# 1103200034

# Machine Learning Class

# Telkom University

In [125]:
# Import library Google Colab untuk mengakses Google Drive
from google.colab import drive

# Mount (pasang) Google Drive pada direktori /content/drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Kode di atas digunakan dalam lingkungan Google Colab untuk mengakses Google Drive. Berikut penjelasan dari setiap baris kode:

1. `from google.colab import drive`: Ini adalah pernyataan impor yang mengimpor modul `drive` dari pustaka `google.colab`. Modul ini digunakan untuk mengakses Google Drive dari lingkungan Google Colab.

2. `drive.mount('/content/drive')`: Pada baris ini, kita memanggil metode `mount` dari modul `drive` yang telah diimpor sebelumnya. Metode ini digunakan untuk "mount" atau "pasang" Google Drive ke dalam direktori tertentu dalam lingkungan Google Colab. Di sini, Google Drive akan di-"mount" pada direktori `/content/drive`.

Dengan kata lain, kode ini menghubungkan Google Drive pengguna dengan lingkungan Google Colab, sehingga pengguna dapat mengakses dan mengelola file dan folder di Google Drive mereka langsung dari lingkungan Colab tersebut. Setelah menjalankan kode ini, Anda akan diminta untuk memberikan izin dan mengautentikasi akun Google Anda agar dapat mengakses Google Drive dari Colab.

In [126]:
# Import library Pandas dan menggantikan 'pd' sebagai alias untuk penggunaan lebih lanjut
import pandas as pd

Kode di atas adalah contoh penggunaan perintah Python yang mengimpor perpustakaan Pandas ke dalam program. Pandas adalah perpustakaan yang digunakan untuk melakukan analisis dan manipulasi data dalam Python. Pada kode tersebut, perintah `import pandas as pd` digunakan untuk mengimpor perpustakaan Pandas dan memberikan alias "pd" ke perpustakaan tersebut. Alias ini digunakan untuk memudahkan penggunaan Pandas di seluruh program, karena dengan menggunakan "pd," kita dapat mengakses berbagai fungsi dan objek Pandas tanpa harus mengetik "pandas" secara penuh setiap kali.

Dengan kata lain, kode di atas mengimpor Pandas dan menggantikan "pandas" dengan "pd" sehingga kita dapat mengakses fungsi-fungsi Pandas dengan lebih mudah, misalnya dengan menuliskan `pd.read_csv()` untuk membaca file CSV atau `pd.DataFrame()` untuk membuat kerangka data (dataframe) Pandas.

In [127]:
#Membaca file CSV yang berisi data ke dalam sebuah DataFrame
data = pd.read_csv('/content/drive/MyDrive/Dataset/mcdonalds_dataset.csv')

Kode di atas digunakan untuk membaca file CSV yang berisi data dan memuatnya ke dalam sebuah DataFrame menggunakan library Python pandas. Berikut adalah penjelasan dari setiap baris kode:

1. `data = pd.read_csv('/content/drive/MyDrive/Dataset/mcdonalds_dataset.csv')`: Pada baris ini, kita menggunakan modul `pd` yang sebelumnya telah diimpor (biasanya sebagai `import pandas as pd`) untuk membaca file CSV. Fungsi `read_csv` dari pandas digunakan untuk membaca data dari file CSV. Dalam kasus ini, alamat file CSV yang akan dibaca adalah `'/content/drive/MyDrive/Dataset/mcdonalds_dataset.csv'`.

   Hasil dari fungsi `read_csv` akan menjadi DataFrame yang berisi data dari file CSV, dan DataFrame ini disimpan dalam variabel `data`. Sehingga, kita dapat mengakses dan memanipulasi data tersebut menggunakan variabel `data`.

In [128]:
#Menampilkan isi dari dataset
data

Unnamed: 0                                    url_for_product  \
0             0   https://mcdonalds-menu.co.uk/bacon-mayo-chicken/   
1             1         https://mcdonalds-menu.co.uk/mayo-chicken/   
2             2            https://mcdonalds-menu.co.uk/hamburger/   
3             3                https://mcdonalds-menu.co.uk/fries/   
4             4         https://mcdonalds-menu.co.uk/cheeseburger/   
..          ...                                                ...   
147         147      https://mcdonalds-menu.co.uk/muffin-with-jam/   
148         148           https://mcdonalds-menu.co.uk/hash-brown/   
149         149       https://mcdonalds-menu.co.uk/pancakes-syrup/   
150         150  https://mcdonalds-menu.co.uk/pancakes-sausage-...   
151         151             https://mcdonalds-menu.co.uk/porridge/   

                      product_name  product_availability product_calories  \
0               Bacon Mayo Chicken                     1        kcal: 332   
1                     Mayo Chicken                     1        kcal: 285   
2                        Hamburger                     1        kcal: 251   
3                            Fries                     1        kcal: 237   
4                     Cheeseburger                     1        kcal: 298   
..                             ...                   ...              ...   
147                Muffin with Jam                     1        kcal: 214   
148                     Hash Brown                     1        kcal: 127   
149               Pancakes & Syrup                     1        kcal: 477   
150  Pancakes & Sausage with Syrup                     1        kcal: 612   
151                       Porridge                     1        kcal: 154   

    product_price  have_sizes  Protein  Carbs   Fat  Salt  Saturates  Sugars  
0          £ 1.99           1     17.0   38.0  12.0  1.40        2.2     4.6  
1          £ 1.39           1     12.0   38.0   9.1  1.00        1.1     4.4  
2           £1.19           1     13.0   30.0   8.3  1.20        3.3     6.3  
3          £ 1.19           3      2.3   29.0  12.0  0.44        1.1     0.4  
4           £1.39           1     16.0   31.0  12.0  1.70        5.5     7.0  
..            ...         ...      ...    ...   ...   ...        ...     ...  
147        £ 1.49           1      5.5   35.0   5.6  0.44        1.4    11.0  
148        £ 1.19           1      1.1   13.0   7.4  0.64        0.7     0.2  
149        £ 3.29           1     10.0   95.0   5.6  1.50        0.9    41.0  
150        £ 3.29           1     20.0   96.0  16.0  2.30        4.6    41.0  
151        £ 1.99           1      7.8   23.0   3.0  0.17        1.3     6.4  

[152 rows x 13 columns]

Kode di atas adalah perintah dalam bahasa pemrograman Python untuk menampilkan isi dari suatu dataset yang disebut "data." Dataset ini terdiri dari 152 baris dan 13 kolom. Berikut adalah penjelasan singkat untuk setiap kolom dataset ini:

1. `Unnamed: 0`: Ini adalah indeks baris dataset yang tidak memiliki label kolom. Ini adalah nomor baris dari 0 hingga 151.

2. `url_for_product`: Kolom ini berisi URL produk makanan dari situs web mcdonalds-menu.co.uk. Ini adalah tautan ke halaman produk tertentu.

3. `product_name`: Ini adalah nama produk makanan yang tersedia di menu McDonald's.

4. `product_availability`: Menunjukkan ketersediaan produk (1 berarti tersedia).

5. `product_calories`: Berisi informasi tentang jumlah kalori dalam produk makanan. Nilai ini disajikan dalam format "kcal: X" di mana X adalah jumlah kalori.

6. `product_price`: Harga produk dalam poundsterling (£).

7. `have_sizes`: Menunjukkan apakah produk ini memiliki opsi ukuran (1 berarti ada).

8. `Protein`: Kandungan protein dalam produk makanan.

9. `Carbs`: Kandungan karbohidrat dalam produk makanan.

10. `Fat`: Kandungan lemak dalam produk makanan.

11. `Salt`: Kandungan garam dalam produk makanan.

12. `Saturates`: Kandungan lemak jenuh dalam produk makanan.

13. `Sugars`: Kandungan gula dalam produk makanan.

Dataset ini mungkin digunakan untuk menganalisis nutrisi, harga, atau ketersediaan produk makanan di menu McDonald's berdasarkan informasi yang diberikan dalam kolom-kolom tersebut.

In [129]:
# Bersihkan kolom 'product_price' dengan menghapus semua karakter yang bukan angka atau desimal
data['product_price'] = data['product_price'].str.replace('[^\d.]', '', regex=True)

# Konversi kolom 'product_price' yang sudah dibersihkan menjadi tipe data float
data['product_price'] = data['product_price'].astype(float)

# Periksa tipe data semua kolom untuk memastikan mereka adalah numerik (kecuali URL dan nama produk)
data.dtypes



Unnamed: 0                int64
url_for_product          object
product_name             object
product_availability      int64
product_calories         object
product_price           float64
have_sizes                int64
Protein                 float64
Carbs                   float64
Fat                     float64
Salt                    float64
Saturates               float64
Sugars                  float64
dtype: object

Kode di atas digunakan untuk membersihkan kolom 'product_price' dalam suatu DataFrame (disebut 'data') dan mengubahnya menjadi tipe data float. Berikut penjelasan dari setiap baris kode:

1. `data['product_price'] = data['product_price'].str.replace('[^\d.]', '', regex=True)`:
   - Pada baris pertama, kita mengambil kolom 'product_price' dari DataFrame 'data'.
   - Kemudian, kita menggunakan metode `str.replace()` pada kolom ini. Metode ini digunakan untuk mengganti teks dalam setiap sel data dengan teks baru.
   - Pada parameter pertama metode `str.replace()`, kita menggunakan regular expression `[^\d.]` untuk mencari semua karakter yang bukan angka (`\d`) atau tanda desimal (`.`).
   - Pada parameter kedua metode `str.replace()`, kita menggantikan karakter-karakter yang ditemukan dengan string kosong (`''`), yang artinya kita menghapus karakter-karakter tersebut dari setiap sel dalam kolom 'product_price'.
   - `regex=True` digunakan untuk mengaktifkan penggunaan regular expression.

2. `data['product_price'] = data['product_price'].astype(float)`:
   - Pada baris kedua, setelah membersihkan kolom 'product_price' dari karakter-karakter yang tidak diinginkan, kita mengubah seluruh isi kolom ini menjadi tipe data float.
   - Metode `astype(float)` digunakan untuk melakukan konversi tipe data.

3. `data.dtypes`:
   - Baris terakhir digunakan untuk menampilkan tipe data dari semua kolom dalam DataFrame 'data' setelah proses pembersihan dan konversi. Tujuannya adalah untuk memastikan bahwa semua kolom, kecuali kolom yang berisi URL dan nama produk, memiliki tipe data numerik, khususnya tipe data float dalam hal ini.

Dengan demikian, kode ini membantu memastikan bahwa kolom 'product_price' dalam DataFrame 'data' sekarang berisi nilai numerik yang siap digunakan untuk analisis data lebih lanjut.

In [130]:
# Ini adalah kode untuk menampilkan informasi tentang objek 'data'
data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 152 entries, 0 to 151
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            152 non-null    int64  
 1   url_for_product       134 non-null    object 
 2   product_name          152 non-null    object 
 3   product_availability  152 non-null    int64  
 4   product_calories      152 non-null    object 
 5   product_price         152 non-null    float64
 6   have_sizes            152 non-null    int64  
 7   Protein               132 non-null    float64
 8   Carbs                 133 non-null    float64
 9   Fat                   133 non-null    float64
 10  Salt                  132 non-null    float64
 11  Saturates             132 non-null    float64
 12  Sugars                129 non-null    float64
dtypes: float64(7), int64(3), object(3)
memory usage: 15.6+ KB


Kode di atas merupakan contoh dari pemanggilan metode `info()` pada sebuah objek DataFrame yang berasal dari pustaka Pandas dalam bahasa pemrograman Python. Kode ini digunakan untuk menampilkan informasi tentang struktur dan karakteristik dari objek DataFrame yang disebut 'data'. Berikut penjelasan dari hasil output yang diberikan oleh metode `info()`:

1. `class 'pandas.core.frame.DataFrame'`: Ini menunjukkan bahwa objek 'data' adalah sebuah DataFrame, yaitu struktur data tabular yang digunakan untuk menyimpan dan mengelola data dalam bentuk tabel.

2. `RangeIndex: 152 entries, 0 to 151`: Ini menunjukkan bahwa DataFrame 'data' memiliki 152 baris (entries) dengan indeks yang berjalan dari 0 hingga 151. RangeIndex adalah jenis indeks bawaan yang digunakan jika tidak ada indeks yang diberikan secara eksplisit.

3. `Data columns (total 13 columns)`: Ini memberikan informasi bahwa DataFrame 'data' terdiri dari 13 kolom (columns).

4. Informasi pada setiap kolom:
   - `Unnamed: 0`: Terdapat 152 data non-null dengan tipe data int64.
   - `url_for_product`: Terdapat 134 data non-null dengan tipe data object (string).
   - `product_name`: Terdapat 152 data non-null dengan tipe data object (string).
   - `product_availability`: Terdapat 152 data non-null dengan tipe data int64.
   - `product_calories`: Terdapat 152 data non-null dengan tipe data object (string).
   - `product_price`: Terdapat 152 data non-null dengan tipe data float64.
   - `have_sizes`: Terdapat 152 data non-null dengan tipe data int64.
   - `Protein`: Terdapat 132 data non-null dengan tipe data float64.
   - `Carbs`: Terdapat 133 data non-null dengan tipe data float64.
   - `Fat`: Terdapat 133 data non-null dengan tipe data float64.
   - `Salt`: Terdapat 132 data non-null dengan tipe data float64.
   - `Saturates`: Terdapat 132 data non-null dengan tipe data float64.
   - `Sugars`: Terdapat 129 data non-null dengan tipe data float64.

5. `dtypes`: Ini memberikan informasi tentang tipe data dari setiap kolom dalam DataFrame, yaitu terdapat 7 kolom dengan tipe data float64, 3 kolom dengan tipe data int64, dan 3 kolom dengan tipe data object (string).

6. `memory usage: 15.6+ KB`: Ini menunjukkan perkiraan penggunaan memori yang digunakan oleh DataFrame 'data', sekitar 15.6 kilobyte.

In [131]:
# Menghitung jumlah nilai NaN (Not a Number) pada kolom 'Protein'
nan_count_Protein = data['Protein'].isna().sum()

# Menghitung jumlah nilai NaN pada kolom ' Carbs  '
nan_count_Carbs = data['Carbs'].isna().sum()

# Menghitung jumlah nilai NaN pada kolom 'Fat'
nan_count_Fat = data['Fat'].isna().sum()

# Menghitung jumlah nilai NaN pada kolom 'Salt'
nan_count_Salt = data['Salt'].isna().sum()

# Menghitung jumlah nilai NaN pada kolom ' Saturates'
nan_count_Saturates = data['Saturates'].isna().sum()

# Menghitung jumlah nilai NaN pada kolom 'Sugars'
nan_count_Sugars = data['Sugars'].isna().sum()

# Menampilkan jumlah NaN pada kolom 'Protein' ke layar
print(f"Jumlah NaN pada kolom Protein: {nan_count_Protein}")

# Menampilkan jumlah NaN pada kolom 'Carbs' ke layar
print(f"Jumlah NaN pada kolom Carbs: {nan_count_Carbs}")

# Menampilkan jumlah NaN pada kolom 'Fat' ke layar
print(f"Jumlah NaN pada kolom Fat: {nan_count_Fat}")

# Menampilkan jumlah NaN pada kolom 'Salt' ke layar
print(f"Jumlah NaN pada kolom Salt: {nan_count_Salt}")

# Menampilkan jumlah NaN pada kolom 'Bedroom2' ke layar
print(f"Jumlah NaN pada kolom Saturates: {nan_count_Saturates}")

# Menampilkan jumlah NaN pada kolom 'Bedroom2' ke layar
print(f"Jumlah NaN pada kolom Sugars: {nan_count_Sugars}")


Jumlah NaN pada kolom Protein: 20
Jumlah NaN pada kolom Carbs: 19
Jumlah NaN pada kolom Fat: 19
Jumlah NaN pada kolom Salt: 20
Jumlah NaN pada kolom Saturates: 20
Jumlah NaN pada kolom Sugars: 23


Kode di atas digunakan untuk menghitung jumlah nilai NaN (Not a Number) dalam berbagai kolom data tertentu, dan kemudian menampilkan jumlah NaN tersebut ke layar. Di bawah ini adalah penjelasan singkat tentang setiap bagian kode:

1. `nan_count_Protein = data['Protein'].isna().sum()`: Ini adalah perintah untuk menghitung jumlah nilai NaN dalam kolom 'Protein' dari DataFrame yang disebut 'data'. Hasil perhitungan disimpan dalam variabel 'nan_count_Protein'.

2. `nan_count_Carbs = data['Carbs'].isna().sum()`: Ini adalah perintah yang serupa dengan yang pertama, tetapi kali ini digunakan untuk menghitung jumlah nilai NaN dalam kolom 'Carbs' dan hasilnya disimpan dalam variabel 'nan_count_Carbs'.

3. `nan_count_Fat = data['Fat'].isna().sum()`: Ini adalah perintah yang serupa dengan yang pertama, tetapi digunakan untuk menghitung jumlah nilai NaN dalam kolom 'Fat' dan hasilnya disimpan dalam variabel 'nan_count_Fat'.

4. `nan_count_Salt = data['Salt'].isna().sum()`: Ini adalah perintah yang serupa dengan yang pertama, tetapi digunakan untuk menghitung jumlah nilai NaN dalam kolom 'Salt' dan hasilnya disimpan dalam variabel 'nan_count_Salt'.

5. `nan_count_Saturates = data['Saturates'].isna().sum()`: Ini adalah perintah yang serupa dengan yang pertama, tetapi digunakan untuk menghitung jumlah nilai NaN dalam kolom 'Saturates' dan hasilnya disimpan dalam variabel 'nan_count_Saturates'.

6. `nan_count_Sugars = data['Sugars'].isna().sum()`: Ini adalah perintah yang serupa dengan yang pertama, tetapi digunakan untuk menghitung jumlah nilai NaN dalam kolom 'Sugars' dan hasilnya disimpan dalam variabel 'nan_count_Sugars'.

7. `print(f"Jumlah NaN pada kolom Protein: {nan_count_Protein}")`: Ini adalah perintah untuk mencetak jumlah nilai NaN dalam kolom 'Protein' ke layar.

8. `print(f"Jumlah NaN pada kolom Carbs: {nan_count_Carbs}")`: Ini adalah perintah untuk mencetak jumlah nilai NaN dalam kolom 'Carbs' ke layar.

9. `print(f"Jumlah NaN pada kolom Fat: {nan_count_Fat}")`: Ini adalah perintah untuk mencetak jumlah nilai NaN dalam kolom 'Fat' ke layar.

10. `print(f"Jumlah NaN pada kolom Salt: {nan_count_Salt}")`: Ini adalah perintah untuk mencetak jumlah nilai NaN dalam kolom 'Salt' ke layar.

11. `print(f"Jumlah NaN pada kolom Saturates: {nan_count_Saturates}")`: Ini adalah perintah untuk mencetak jumlah nilai NaN dalam kolom 'Saturates' ke layar.

12. `print(f"Jumlah NaN pada kolom Sugars: {nan_count_Sugars}")`: Ini adalah perintah untuk mencetak jumlah nilai NaN dalam kolom 'Sugars' ke layar.

Hasil dari eksekusi kode ini adalah jumlah nilai NaN yang ditemukan dalam masing-masing kolom yang disebutkan, yang akan dicetak ke layar. Jumlah NaN adalah informasi penting dalam analisis data, karena dapat mengindikasikan data yang hilang atau perlu diimputasi.

In [132]:
# Menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam kolom 'Protein'
data.dropna(subset=['Protein'], inplace=True)

# Menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam kolom 'Carbs'
data.dropna(subset=['Carbs'], inplace=True)

# Menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam kolom 'Fat'
data.dropna(subset=['Fat'], inplace=True)

# Menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam kolom 'Salt'
data.dropna(subset=['Salt'], inplace=True)

# Menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam kolom 'Saturates'
data.dropna(subset=['Saturates'], inplace=True)

# Menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam kolom 'Sugars'
data.dropna(subset=['Sugars'], inplace=True)

Kode di atas adalah contoh kode yang digunakan untuk menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam beberapa kolom tertentu dalam suatu dataset. Kode tersebut dilakukan untuk berbagai kolom berbeda secara berurutan, yaitu 'Protein', 'Carbs', 'Fat', 'Salt', 'Saturates', dan 'Sugars'. Setiap bagian kode memiliki tujuan yang sama, yaitu:

1. `data.dropna(subset=['Protein'], inplace=True)`: Ini menghapus baris dalam dataset 'data' di mana kolom 'Protein' memiliki nilai yang hilang (NaN). `dropna` adalah metode yang digunakan untuk menghapus baris dengan nilai NaN, dan `subset=['Protein']` mengidentifikasi kolom yang harus diperiksa. Parameter `inplace=True` mengindikasikan bahwa perubahan harus diterapkan langsung pada dataset yang ada.

2. Kode yang sama seperti di atas diulang untuk kolom 'Carbs', 'Fat', 'Salt', 'Saturates', dan 'Sugars'. Setiap baris kode ini memiliki tujuan yang sama, yaitu menghilangkan baris-baris dengan nilai NaN dalam kolom yang sesuai.

In [133]:
# Menghitung jumlah nilai NaN (missing values) dalam kolom 'Car' dari DataFrame 'data'
nan_count_Protein = data['Protein'].isna().sum()

# Menghitung jumlah nilai NaN (missing values) dalam kolom 'Carbs' dari DataFrame 'data'
nan_count_Carbs = data['Carbs'].isna().sum()

# Menghitung jumlah nilai NaN (missing values) dalam kolom 'Fat' dari DataFrame 'data'
nan_count_Fat = data['Fat'].isna().sum()

# Menghitung jumlah nilai NaN (missing values) dalam kolom 'Salt' dari DataFrame 'data'
nan_count_Salt = data['Salt'].isna().sum()

# Menghitung jumlah nilai NaN (missing values) dalam kolom 'Saturates' dari DataFrame 'data'
nan_count_Saturates = data['Saturates'].isna().sum()

# Menghitung jumlah nilai NaN (missing values) dalam kolom 'Sugars' dari DataFrame 'data'
nan_count_Sugars = data['Sugars'].isna().sum()


# Menampilkan jumlah nilai NaN dalam kolom 'Protein' menggunakan f-string
print(f"Jumlah nilai NaN dalam kolom Protein: {nan_count_Protein}")

# Menampilkan jumlah nilai NaN dalam kolom 'Carbs' menggunakan f-string
print(f"Jumlah nilai NaN dalam kolom Carbs: {nan_count_Carbs}")

# Menampilkan jumlah nilai NaN dalam kolom 'Fat' menggunakan f-string
print(f"Jumlah nilai NaN dalam kolom Fat: {nan_count_Fat}")

# Menampilkan jumlah nilai NaN dalam kolom 'Salt' menggunakan f-string
print(f"Jumlah nilai NaN dalam kolom Salt: {nan_count_Salt}")

# Menampilkan jumlah nilai NaN dalam kolom 'Saturates' menggunakan f-string
print(f"Jumlah nilai NaN dalam kolom Saturates: {nan_count_Saturates}")

# Menampilkan jumlah nilai NaN dalam kolom 'Sugars' menggunakan f-string
print(f"Jumlah nilai NaN dalam kolom Sugars: {nan_count_Sugars}")

Jumlah nilai NaN dalam kolom Protein: 0
Jumlah nilai NaN dalam kolom Carbs: 0
Jumlah nilai NaN dalam kolom Fat: 0
Jumlah nilai NaN dalam kolom Salt: 0
Jumlah nilai NaN dalam kolom Saturates: 0
Jumlah nilai NaN dalam kolom Sugars: 0


Kode di atas adalah contoh penggunaan Python dan library pandas untuk menghitung jumlah nilai NaN (missing values) dalam beberapa kolom tertentu dari sebuah DataFrame yang disebut 'data'. Berikut adalah penjelasan dari kode tersebut:

1. `nan_count_Protein = data['Protein'].isna().sum()`: Dalam baris ini, kita menghitung jumlah nilai NaN dalam kolom 'Protein' dari DataFrame 'data' dengan menggunakan metode `.isna()` untuk mengidentifikasi nilai-nilai yang hilang (NaN) dalam kolom 'Protein', dan kemudian kita menggunakan `.sum()` untuk menghitung total jumlah nilai NaN dalam kolom tersebut.

2. `nan_count_Carbs = data['Carbs'].isna().sum()`: Langkah ini sama seperti langkah pertama, tetapi kali ini kita menghitung jumlah nilai NaN dalam kolom 'Carbs'.

3. `nan_count_Fat = data['Fat'].isna().sum()`: Seperti langkah sebelumnya, di sini kita menghitung jumlah nilai NaN dalam kolom 'Fat'.

4. `nan_count_Salt = data['Salt'].isna().sum()`: Sama seperti sebelumnya, ini menghitung jumlah nilai NaN dalam kolom 'Salt'.

5. `nan_count_Saturates = data['Saturates'].isna().sum()`: Ini menghitung jumlah nilai NaN dalam kolom 'Saturates'.

6. `nan_count_Sugars = data['Sugars'].isna().sum()`: Langkah terakhir ini menghitung jumlah nilai NaN dalam kolom 'Sugars'.

Setelah menghitung jumlah nilai NaN dalam masing-masing kolom, kode kemudian menggunakan f-string untuk mencetak hasilnya. Misalnya, `print(f"Jumlah nilai NaN dalam kolom Protein: {nan_count_Protein}")` digunakan untuk mencetak jumlah nilai NaN dalam kolom 'Protein', dan langkah serupa dilakukan untuk kolom lainnya. Dalam contoh yang diberikan, semua kolom tersebut memiliki jumlah nilai NaN yang sama, yaitu 0, yang berarti tidak ada nilai NaN dalam kolom tersebut.

In [134]:
# Mengambil bentuk (shape) dari data
data.shape


(129, 13)

Kode di atas digunakan untuk mengambil bentuk (shape) dari data yang disimpan dalam suatu variabel yang tidak ditampilkan di dalam pertanyaan. Hasil dari kode tersebut adalah (129, 13), yang mengindikasikan bahwa data tersebut memiliki 129 baris dan 13 kolom.

In [135]:
# Mengubah kolom 'Protein' menjadi tipe data integer
data['Protein'] = data['Protein'].astype(int)

# Mengubah kolom 'Carbs' menjadi tipe data integer
data['Carbs'] = data['Carbs'].astype(int)

# Mengubah kolom 'Fat' menjadi tipe data integer
data['Fat'] = data['Fat'].astype(int)

# Mengubah kolom 'Salt' menjadi tipe data integer
data['Salt'] = data['Salt'].astype(int)

# Mengubah kolom 'Saturates' menjadi tipe data integer
data['Saturates'] = data['Saturates'].astype(int)

# Mengubah kolom 'Sugars' menjadi tipe data integer
data['Sugars'] = data['Sugars'].astype(int)

Kode di atas digunakan untuk mengubah tipe data dari beberapa kolom dalam sebuah DataFrame (mungkin dari library seperti pandas) menjadi tipe data integer. Ini dilakukan dengan menggunakan metode `.astype(int)` pada setiap kolom yang diinginkan, dan kolom yang diubah termasuk 'Protein', 'Carbs', 'Fat', 'Salt', 'Saturates', dan 'Sugars'. Hasilnya adalah bahwa semua nilai dalam kolom-kolom ini akan menjadi bilangan bulat (integer) daripada tipe data sebelumnya yang mungkin merupakan tipe data lain seperti float atau string.

In [136]:
# Pilih subset dari predictor (variabel independen) yang akan digunakan dalam analisis.
cols_to_use = ['product_availability', 'have_sizes', 'Protein', 'Carbs', 'Fat', 'Salt', 'Saturates', 'Sugars']

# Buat dataframe 'X' yang hanya berisi kolom-kolom yang terdapat dalam 'cols_to_use'.
X = data[cols_to_use]


Kode di atas adalah bagian dari sebuah analisis data yang menggunakan Python, mungkin dalam konteks ilmu data atau statistik. Kode tersebut melakukan langkah-langkah berikut:

1. Terlebih dahulu, ada sebuah list bernama `cols_to_use` yang berisi nama-nama kolom (predictor atau variabel independen) yang akan digunakan dalam analisis selanjutnya. Kolom-kolom ini tampaknya berasal dari sebuah dataframe yang diberi nama `data`.

2. Kemudian, kode menciptakan sebuah dataframe baru yang disebut `X`. Dataframe ini hanya akan berisi kolom-kolom yang terdapat dalam list `cols_to_use`. Dengan kata lain, kode ini memilih subset dari predictor yang ingin digunakan dalam analisis dan menyimpannya dalam dataframe `X` untuk penggunaan selanjutnya.

Dengan menggunakan subset ini, analisis selanjutnya mungkin akan fokus pada hubungan atau pola yang mungkin ada antara variabel-variabel ini dengan variabel target atau tujuan analisis.

In [137]:
# Select target
y = data.product_price


Kode di atas merupakan potongan kode dalam bahasa pemrograman Python yang digunakan untuk mengatur variabel target dalam suatu analisis data atau pemodelan statistik/machine learning. Dalam kode tersebut:

- `y`: Ini adalah nama variabel yang digunakan untuk menyimpan target data atau variabel yang ingin diprediksi. Dalam kasus ini, targetnya adalah `product_price` yang merupakan harga produk. Variabel `y` akan digunakan dalam berbagai analisis data atau pemodelan untuk memprediksi atau menganalisis hubungan dengan fitur-fitur lainnya.

Namun, untuk memberikan penjelasan lebih detail, perlu informasi tambahan tentang konteks kode tersebut, seperti bagaimana variabel `data` dan `product_price` didefinisikan dan digunakan dalam konteks yang lebih luas.

In [138]:
from sklearn.model_selection import train_test_split

# Memisahkan data menjadi set pelatihan dan validasi
X_train, X_valid, y_train, y_valid = train_test_split(X, y, random_state=86, test_size=0.100)


Kode di atas mengimport fungsi `train_test_split` dari pustaka `sklearn.model_selection`. Fungsi ini digunakan untuk memisahkan dataset menjadi dua bagian, yaitu set pelatihan (train set) dan set validasi (validation set), sehingga kita dapat menguji kinerja model pada data yang tidak digunakan dalam proses pelatihan.

Penjelasan lebih rinci dari kode tersebut adalah sebagai berikut:
- `X_train` adalah variabel yang akan menyimpan fitur (features) dari set pelatihan.
- `X_valid` adalah variabel yang akan menyimpan fitur dari set validasi.
- `y_train` adalah variabel yang akan menyimpan target (labels) dari set pelatihan.
- `y_valid` adalah variabel yang akan menyimpan target dari set validasi.

Fungsi `train_test_split` menerima beberapa argumen:
- `X` adalah matriks fitur lengkap.
- `y` adalah vektor target yang sesuai dengan matriks fitur.
- `random_state` adalah argumen yang digunakan untuk mengontrol randomization ketika memisahkan data. Dengan mengatur nilai ini, Anda dapat memastikan bahwa pemisahan data akan konsisten jika Anda perlu menjalankan kode ini beberapa kali.
- `test_size` adalah argumen yang menentukan seberapa besar proporsi data yang akan dialokasikan untuk set validasi. Dalam kasus ini, 0.10 (atau 10%) dari data akan digunakan sebagai set validasi.

Setelah menjalankan kode ini, Anda akan memiliki empat variabel yang berisi data yang telah dipisahkan dengan cara yang sesuai untuk pelatihan dan validasi model machine learning.

In [139]:
# Mengidentifikasi kolom dengan data string
string_columns = ['product_availability', 'have_sizes', 'Protein', 'Carbs', 'Fat', 'Salt', 'Saturates', 'Sugars']

# Mengkonversi kolom yang memiliki data string menjadi tipe data numerik (contoh: one-hot encoding)
X_train_encoded = pd.get_dummies(X_train, columns=string_columns)

Kode di atas bertujuan untuk mengidentifikasi kolom-kolom dalam sebuah DataFrame yang memiliki data string, kemudian mengkonversi kolom-kolom tersebut menjadi tipe data numerik dengan teknik one-hot encoding. Inilah penjelasan lebih detailnya:

1. `string_columns` adalah sebuah daftar (list) yang berisi nama-nama kolom yang terdapat dalam DataFrame `X_train` dan memiliki data string. Kolom-kolom ini mungkin berisi informasi seperti teks atau kategori yang perlu diubah menjadi bentuk numerik agar dapat digunakan dalam model pemodelan data.

2. `pd.get_dummies(X_train, columns=string_columns)` adalah panggilan fungsi dari pustaka pandas (disingkat sebagai `pd`) yang digunakan untuk melakukan one-hot encoding pada DataFrame `X_train`. One-hot encoding adalah teknik yang digunakan untuk mengubah kolom-kolom kategori menjadi kolom-kolom biner (0 dan 1) yang merepresentasikan keberadaan atau ketiadaan setiap kategori.

   - `X_train` adalah DataFrame yang akan diubah.
   - `columns=string_columns` adalah argumen yang digunakan untuk menentukan kolom-kolom mana yang akan diubah menjadi one-hot encoding. Kolom-kolom yang terdapat dalam `string_columns` akan diubah, sedangkan kolom-kolom lainnya akan tetap dalam bentuk aslinya.

Hasil dari operasi ini adalah DataFrame baru yang sudah diubah, di mana kolom-kolom yang terdapat dalam `string_columns` telah di-one-hot encoding, dan kolom-kolom tersebut sekarang berbentuk biner yang merepresentasikan nilai-nilai yang terdapat dalam kolom tersebut. Ini memungkinkan penggunaan data tersebut dalam model-machine learning yang memerlukan data numerik, seperti regresi atau klasifikasi.

In [140]:
# Import library yang diperlukan
from xgboost import XGBRegressor

# Membuat objek model XGBoost
my_model = XGBRegressor()

# Melatih model dengan data pelatihan X_train dan target y_train
my_model.fit(X_train, y_train)


XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=None, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=None, n_jobs=None,
             num_parallel_tree=None, random_state=None, ...)

Kode di atas adalah contoh penggunaan library XGBoost untuk membuat dan melatih model regresi menggunakan `XGBRegressor`. Berikut penjelasan dari kode tersebut:

1. Baris pertama mengimpor library yang diperlukan, yaitu `XGBRegressor` dari library XGBoost. Library ini digunakan untuk membangun model regresi dengan algoritma XGBoost.

2. Baris kedua menciptakan objek model `my_model` menggunakan `XGBRegressor()`. Ini adalah langkah awal untuk membuat model regresi yang akan digunakan untuk memprediksi data.

3. Baris ketiga melatih model dengan data pelatihan. `my_model.fit(X_train, y_train)` akan melatih model dengan data yang disimpan dalam `X_train` sebagai fitur (features) dan `y_train` sebagai target (target values). Hasil dari pelatihan ini adalah model XGBoost yang siap digunakan untuk membuat prediksi.

4. Di bawah kode tersebut, terdapat informasi tentang parameter yang dapat dikonfigurasi ketika membuat objek `XGBRegressor`. Ini mencakup berbagai parameter yang dapat disesuaikan, seperti learning rate, jumlah pohon (n_estimators), kedalaman maksimum pohon (max_depth), dan banyak parameter lainnya yang memengaruhi perilaku model XGBoost.

Kode di atas adalah contoh dasar untuk melatih model regresi menggunakan XGBoost. Untuk penggunaan yang lebih kompleks, biasanya akan melakukan penyetelan parameter dan evaluasi model lebih lanjut.

In [141]:
# Mengimpor modul mean_absolute_error dari sklearn.metrics
from sklearn.metrics import mean_absolute_error

# Melakukan prediksi menggunakan model (my_model) pada data validasi (X_valid)
predictions = my_model.predict(X_valid)

# Menghitung Mean Absolute Error (MAE) antara prediksi dan nilai target aktual (y_valid)
# dan mencetak hasilnya
print("Mean Absolute Error: " + str(mean_absolute_error(predictions, y_valid)))


Mean Absolute Error: 0.24468636732835045


Kode di atas adalah contoh penggunaan modul `mean_absolute_error` dari pustaka scikit-learn (sklearn) dalam bahasa pemrograman Python. Berikut penjelasan singkat dari setiap baris kode:

1. Import modul `mean_absolute_error` dari `sklearn.metrics`:
   ```python
   from sklearn.metrics import mean_absolute_error
   ```
   Dalam baris ini, kita mengimpor fungsi `mean_absolute_error` dari pustaka scikit-learn (sklearn) yang akan digunakan untuk menghitung Mean Absolute Error (MAE) antara prediksi dan nilai target aktual.

2. Melakukan prediksi menggunakan model (my_model) pada data validasi (X_valid):
   ```python
   predictions = my_model.predict(X_valid)
   ```
   Di sini, model yang disimpan dalam variabel `my_model` digunakan untuk membuat prediksi pada data validasi yang disimpan dalam variabel `X_valid`. Hasil prediksi disimpan dalam variabel `predictions`.

3. Menghitung Mean Absolute Error (MAE) antara prediksi dan nilai target aktual (y_valid):
   ```python
   print("Mean Absolute Error: " + str(mean_absolute_error(predictions, y_valid)))
   ```
   Baris ini mencetak MAE antara prediksi yang telah dihasilkan oleh model dan nilai target aktual yang disimpan dalam variabel `y_valid`. Hasil MAE ditampilkan sebagai teks di layar.

Hasil yang dicetak merupakan nilai MAE yang mengukur seberapa besar kesalahan rata-rata (dalam nilai absolut) antara prediksi model dengan data target aktual. Semakin kecil nilai MAE, semakin baik kinerja model dalam melakukan prediksi. Nilai MAE sekitar 0.2447 dalam contoh ini, yang menunjukkan bahwa model memiliki kesalahan rata-rata sekitar 0.2447 unit dari nilai target aktual.

In [142]:
# Membuat model XGBoost Regressor dengan 1000 pohon keputusan (n_estimators=1000)
my_model_0 = XGBRegressor(n_estimators=1000)

# Melatih model menggunakan data pelatihan X_train dan label y_train
my_model_0.fit(X_train, y_train)


XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=None, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=1000, n_jobs=None,
             num_parallel_tree=None, random_state=None, ...)

Kode di atas adalah contoh penggunaan XGBoost Regressor dalam pemodelan machine learning dengan beberapa parameter yang disetel:

1. `my_model_0 = XGBRegressor(n_estimators=1000)`: Di sini, kita membuat sebuah objek model `my_model_0` menggunakan kelas `XGBRegressor` dari pustaka XGBoost. Parameter `n_estimators` diatur menjadi 1000, yang berarti model ini akan menggunakan 1000 pohon keputusan dalam ansambelnya. Pohon keputusan adalah komponen dasar dari model XGBoost yang digunakan untuk melakukan prediksi.

2. `my_model_0.fit(X_train, y_train)`: Dalam baris ini, model yang telah dibuat (`my_model_0`) dilatih menggunakan data pelatihan `X_train` dan label `y_train`. Ini adalah tahap pelatihan di mana model belajar dari data pelatihan untuk membuat prediksi yang lebih baik di masa depan. Model XGBoost akan melakukan serangkaian iterasi atau langkah-langkah untuk mengoptimalkan pohon keputusan dan parameter lainnya berdasarkan data pelatihan.

Selain dari itu, kode tersebut juga mencantumkan beberapa parameter dan argumen default lainnya yang tidak diubah dari nilai defaultnya. Ini adalah parameter bawaan XGBoost yang dapat disesuaikan sesuai kebutuhan untuk meningkatkan kinerja model. Contoh parameter yang dapat diubah mencakup tingkat pembelajaran (`learning_rate`), kedalaman maksimum pohon (`max_depth`), dan lainnya.

Jadi, keseluruhan tujuan dari kode di atas adalah untuk membuat dan melatih model XGBoost Regressor dengan 1000 pohon keputusan sebagai bagian dari ansambelnya untuk memprediksi nilai berdasarkan data pelatihan yang diberikan. Model ini akan digunakan untuk tugas regresi, yaitu memprediksi nilai berkelanjutan.

In [143]:
# Melakukan prediksi menggunakan model 'my_model_0' pada data validasi 'X_valid'
predictions_0 = my_model_0.predict(X_valid)

# Menghitung Mean Absolute Error (MAE) antara prediksi dan target sebenarnya 'y_valid'
mae = mean_absolute_error(predictions_0, y_valid)

# Menampilkan MAE ke layar
print("Mean Absolute Error: " + str(mae))


Mean Absolute Error: 0.24468636732835045


Kode di atas melakukan beberapa langkah sebagai berikut:

1. `my_model_0.predict(X_valid)`: Menggunakan model yang disebut "my_model_0", kode ini melakukan prediksi pada data validasi yang disimpan dalam variabel 'X_valid'. Model ini adalah model yang telah dilatih sebelumnya, dan saat ini digunakan untuk melakukan prediksi pada data yang belum dilihat sebelumnya.

2. `mean_absolute_error(predictions_0, y_valid)`: Kode ini menghitung Mean Absolute Error (MAE) antara hasil prediksi yang dihasilkan oleh model ('predictions_0') dan target sebenarnya yang disimpan dalam variabel 'y_valid'. MAE adalah metrik evaluasi yang mengukur seberapa dekat prediksi model dengan nilai sebenarnya. MAE dihitung dengan cara mengambil selisih absolut antara setiap prediksi dan nilai sebenarnya, dan kemudian menghitung rata-rata dari seluruh selisih ini.

3. `print("Mean Absolute Error: " + str(mae))`: Kode ini mencetak hasil MAE yang telah dihitung ke layar, sehingga kita dapat mengetahui seberapa baik model 'my_model_0' dalam melakukan prediksi terhadap data validasi. Dalam contoh ini, MAE yang dicetak adalah sekitar 0.24468636732835045, yang merupakan nilai rata-rata dari selisih absolut antara prediksi dan nilai sebenarnya. Semakin rendah nilai MAE, semakin baik model tersebut dalam melakukan prediksi.

In [144]:
from sklearn.model_selection import GridSearchCV

# Tentukan model XGBoost
model_grid = XGBRegressor()

# Tentukan parameter grid
param_grid = {
    'n_estimators': [250, 300, 450, 500, 1100]  # Ganti dengan rentang nilai yang ingin Anda coba
}

# Inisialisasi GridSearchCV
grid_search = GridSearchCV(estimator=model_grid, param_grid=param_grid, cv=5, scoring='neg_mean_absolute_error')

# Lakukan fitting menggunakan data training
grid_search.fit(X_train, y_train)

# Dapatkan parameter terbaik
best_n_estimators = grid_search.best_params_['n_estimators']

print("Nilai n_estimators terbaik:", best_n_estimators)
# Tentukan range nilai n_estimators yang akan diuji
n_estimators_list = [100, 150, 200, 250, 300]

# Inisialisasi sebuah list untuk menyimpan hasil MAE
mae_results = []

# Loop melalui setiap nilai n_estimators
for n_estimators in n_estimators_list:
    # Inisialisasi model dengan n_estimators tertentu
    model_grid = XGBRegressor(n_estimators=n_estimators)

    # Fitting model menggunakan data training
    model_grid.fit(X_train, y_train)

    # Lakukan prediksi pada data validasi
    predictions = model_grid.predict(X_valid)

    # Hitung Mean Absolute Error
    mae = mean_absolute_error(predictions, y_valid)

    # Tampilkan nilai n_estimators dan MAE
    print(f"n_estimators: {n_estimators}, Mean Absolute Error: {mae}")

    # Tambahkan nilai MAE ke dalam list
    mae_results.append(mae)

Nilai n_estimators terbaik: 250
n_estimators: 100, Mean Absolute Error: 0.24468636732835045
n_estimators: 150, Mean Absolute Error: 0.24468636732835045
n_estimators: 200, Mean Absolute Error: 0.24468636732835045
n_estimators: 250, Mean Absolute Error: 0.24468636732835045
n_estimators: 300, Mean Absolute Error: 0.24468636732835045


Kode di atas adalah contoh penggunaan GridSearchCV dari library scikit-learn (sklearn) untuk mencari nilai terbaik dari parameter 'n_estimators' pada model XGBoost (XGBRegressor) dalam konteks regresi. Berikut adalah penjelasan singkat untuk setiap bagian kode:

1. Import GridSearchCV:
   Baris pertama mengimpor GridSearchCV dari modul `sklearn.model_selection`. GridSearchCV adalah alat untuk melakukan pencarian parameter terbaik dengan berbagai kombinasi parameter yang telah ditentukan.

2. Tentukan model XGBoost:
   Baris kedua menginisialisasi model XGBoost sebagai `model_grid` dengan parameter default.

3. Tentukan parameter grid:
   Baris keempat mendefinisikan parameter grid `param_grid` yang akan digunakan dalam pencarian parameter terbaik. Dalam contoh ini, hanya satu parameter yang akan dicari, yaitu 'n_estimators', dan beberapa nilai yang mungkin.

4. Inisialisasi GridSearchCV:
   Baris keenam menginisialisasi objek GridSearchCV dengan model `model_grid`, parameter grid `param_grid`, metode validasi silang (cross-validation) sebanyak 5 kali (cv=5), dan menggunakan 'neg_mean_absolute_error' sebagai metrik penilaian.

5. Lakukan fitting menggunakan data training:
   Baris kedelapan melakukan fitting model dengan data training `X_train` dan target `y_train` menggunakan GridSearchCV. Ini akan melakukan pencarian parameter terbaik berdasarkan parameter grid yang telah ditentukan.

6. Dapatkan parameter terbaik:
   Baris kesepuluh mengekstrak nilai terbaik dari parameter 'n_estimators' yang ditemukan oleh GridSearchCV dan menyimpannya dalam variabel `best_n_estimators`.

7. Print nilai terbaik:
   Baris ke-12 mencetak nilai terbaik dari parameter 'n_estimators' yang ditemukan oleh GridSearchCV.

Setelah mendapatkan nilai terbaik untuk 'n_estimators', kode melanjutkan dengan mencoba beberapa nilai lain dari 'n_estimators' secara manual dalam loop:

8. Tentukan range nilai n_estimators yang akan diuji:
   Baris ke-15 mendefinisikan daftar `n_estimators_list` yang berisi beberapa nilai yang ingin diuji.

9. Inisialisasi list untuk menyimpan hasil MAE:
   Baris ke-18 membuat list kosong `mae_results` yang akan digunakan untuk menyimpan hasil Mean Absolute Error (MAE) dari setiap iterasi.

10. Loop melalui setiap nilai n_estimators:
    Baris ke-21 hingga ke-37 adalah loop yang iterasi melalui setiap nilai 'n_estimators' dalam daftar `n_estimators_list`. Di setiap iterasi, model XGBoost diinisialisasi dengan nilai 'n_estimators' tertentu, kemudian dilakukan fitting pada data training.

11. Lakukan prediksi pada data validasi:
    Baris ke-29 memprediksi hasil menggunakan model yang telah di-fit pada data validasi `X_valid`.

12. Hitung Mean Absolute Error:
    Baris ke-32 menghitung MAE dengan membandingkan prediksi dengan target sebenarnya (`y_valid`) menggunakan fungsi `mean_absolute_error`.

13. Tampilkan nilai n_estimators dan MAE:
    Baris ke-35 mencetak nilai 'n_estimators' dan MAE untuk setiap iterasi.

14. Tambahkan nilai MAE ke dalam list:
    Baris ke-38 menambahkan nilai MAE ke dalam list `mae_results`.

Jadi, kode tersebut menggambarkan proses pencarian parameter terbaik untuk 'n_estimators' dengan menggunakan GridSearchCV, dan kemudian mengevaluasi beberapa nilai 'n_estimators' secara manual untuk model XGBoost. Hasil MAE dari setiap iterasi manual juga dicetak untuk dianalisis.

In [145]:
# Membuat model XGBoost Regressor dengan 100 pohon keputusan (n_estimators=100)
my_model_2 = XGBRegressor(n_estimators=100)

# Melatih model menggunakan data pelatihan (X_train, y_train) dengan early stopping
# untuk menghentikan pelatihan jika tidak ada perbaikan dalam 5 iterasi
# dan menggunakan data validasi (X_valid, y_valid) untuk evaluasi
my_model_2.fit(X_train, y_train, early_stopping_rounds=5, eval_set=[(X_valid, y_valid)], verbose=False)

# Melakukan prediksi pada data validasi menggunakan model yang telah dilatih
predictions_2 = my_model_2.predict(X_valid)

# Menghitung Mean Absolute Error (MAE) antara prediksi dan nilai sebenarnya pada data validasi
print("Mean Absolute Error: " + str(mean_absolute_error(predictions_2, y_valid)))


Mean Absolute Error: 0.269834254704989


/usr/local/lib/python3.10/dist-packages/xgboost/sklearn.py:885: UserWarning: `early_stopping_rounds` in `fit` method is deprecated for better compatibility with scikit-learn, use `early_stopping_rounds` in constructor or`set_params` instead.
  warnings.warn(


Kode di atas adalah contoh penggunaan model XGBoost Regressor untuk melakukan tugas regresi. Berikut adalah penjelasan singkat dari setiap langkah dalam kode tersebut:

1. `my_model_2 = XGBRegressor(n_estimators=100)`: Di sini, model XGBoost Regressor dibuat dengan 100 pohon keputusan (n_estimators=100). Ini berarti model akan terdiri dari 100 pohon keputusan yang digunakan untuk memprediksi variabel target dalam tugas regresi.

2. `my_model_2.fit(X_train, y_train, early_stopping_rounds=5, eval_set=[(X_valid, y_valid)], verbose=False)`: Model XGBoost dilatih menggunakan data pelatihan `X_train` dan target `y_train`. Parameter `early_stopping_rounds` digunakan untuk menghentikan pelatihan jika tidak ada perbaikan dalam 5 iterasi. Data validasi `(X_valid, y_valid)` digunakan untuk evaluasi model selama pelatihan. `verbose=False` mengatur agar tidak ada output pelatihan yang ditampilkan.

3. `predictions_2 = my_model_2.predict(X_valid)`: Setelah pelatihan selesai, model digunakan untuk melakukan prediksi pada data validasi `X_valid`, dan hasil prediksi disimpan dalam variabel `predictions_2`.

4. `print("Mean Absolute Error: " + str(mean_absolute_error(predictions_2, y_valid)))`: Akhirnya, Mean Absolute Error (MAE) dihitung antara prediksi (`predictions_2`) dan nilai sebenarnya (`y_valid`) pada data validasi. MAE adalah metrik evaluasi yang mengukur seberapa dekat prediksi model dengan nilai sebenarnya, dan hasilnya dicetak ke layar.

Penting untuk diingat bahwa dalam peringatan yang muncul, ada peringatan bahwa parameter `early_stopping_rounds` dalam metode `fit` telah dianggap usang, dan disarankan untuk menggunakannya dalam konstruktor atau `set_params`. Ini adalah perubahan dalam penggunaan XGBoost, tetapi kode di atas masih berfungsi dengan baik untuk tujuan demonstrasi.

In [146]:
# Import library XGBRegressor untuk mengembangkan model XGBoost
my_model_1 = XGBRegressor(n_estimators=100)

# Melatih model menggunakan data pelatihan
my_model_1.fit(X_train, y_train)

# Membuat prediksi menggunakan model yang telah dilatih pada data validasi
predictions_1 = my_model_1.predict(X_valid)

# Menghitung dan mencetak Mean Absolute Error (MAE) dari prediksi model
print("Mean Absolute Error: " + str(mean_absolute_error(predictions_1, y_valid)))


Mean Absolute Error: 0.24468636732835045


Kode di atas adalah contoh implementasi pemodelan menggunakan algoritma XGBoost (Extreme Gradient Boosting) untuk tugas regresi. Berikut penjelasan singkat dari setiap bagian kode:

1. `my_model_1 = XGBRegressor(n_estimators=100)`: Baris ini mengimpor kelas `XGBRegressor` dari pustaka XGBoost dan membuat sebuah objek model regresi yang disebut `my_model_1`. Model ini akan memiliki 100 pohon keputusan sebagai estimator dalam ensemble XGBoost.

2. `my_model_1.fit(X_train, y_train)`: Baris ini melatih model `my_model_1` menggunakan data pelatihan. `X_train` adalah matriks fitur pelatihan, dan `y_train` adalah vektor target pelatihan.

3. `predictions_1 = my_model_1.predict(X_valid)`: Baris ini menggunakan model yang telah dilatih (`my_model_1`) untuk membuat prediksi pada data validasi (`X_valid`). Hasil prediksi disimpan dalam vektor `predictions_1`.

4. `print("Mean Absolute Error: " + str(mean_absolute_error(predictions_1, y_valid)))`: Baris ini menghitung Mean Absolute Error (MAE) antara prediksi model (`predictions_1`) dan target sesungguhnya pada data validasi (`y_valid`). Hasil MAE kemudian dicetak ke layar.

Hasil keluaran dari kode ini adalah MAE, yang merupakan metrik evaluasi untuk mengukur seberapa baik model XGBoost dalam melakukan prediksi terhadap data validasi. Semakin rendah nilai MAE, semakin baik model dalam melakukan prediksi, dengan angka yang lebih rendah menunjukkan bahwa model lebih mendekati nilai target yang sebenarnya. Dalam konteks ini, MAE sebesar 0.24468636732835045 menunjukkan bahwa model memiliki kualitas yang baik dalam melakukan prediksi pada data validasi.

In [147]:
# Membuat model XGBoost Regressor dengan 1000 pohon keputusan (n_estimators=1000).
# Model ini akan digunakan untuk memprediksi variabel target berdasarkan fitur-fitur dalam dataset.
my_model_3 = XGBRegressor(n_estimators=1000, learning_rate=0.10, n_jobs=10)

# Melatih model dengan menggunakan data pelatihan (X_train) dan target (y_train).
my_model_3.fit(X_train, y_train)



XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=0.1, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=1000, n_jobs=10,
             num_parallel_tree=None, random_state=None, ...)

Kode di atas digunakan untuk membuat model XGBoost Regressor dengan 1000 pohon keputusan (n_estimators=1000) dan menggambarkan beberapa parameter serta penggunaan model tersebut. Parameter-parameter yang terlihat dalam output adalah sebagai berikut:

1. `n_estimators=1000`: Ini adalah parameter yang menentukan jumlah pohon keputusan yang akan digunakan dalam model. Dalam kasus ini, model akan menggunakan 1000 pohon keputusan untuk membuat prediksi.

2. `learning_rate=0.10`: Parameter ini mengatur laju pembelajaran (learning rate) yang digunakan oleh model. Laju pembelajaran mempengaruhi sejauh mana model menyesuaikan diri dengan data pelatihan pada setiap iterasi.

3. `n_jobs=10`: Parameter ini mengontrol jumlah pekerjaan paralel yang dapat dilakukan oleh model. Dalam hal ini, model dapat menggunakan hingga 10 pekerjaan secara bersamaan, yang dapat meningkatkan kecepatan pelatihan pada komputer dengan banyak inti CPU.

Selanjutnya, kode tersebut melatih model XGBoost Regressor dengan data pelatihan (X_train) dan target (y_train) menggunakan metode `.fit()`. Setelah pelatihan, model siap digunakan untuk membuat prediksi berdasarkan fitur-fitur dalam dataset. Model ini dapat digunakan untuk masalah regresi, di mana tujuannya adalah memprediksi variabel target numerik berdasarkan fitur-fitur yang diberikan dalam dataset.

In [148]:
# predictions_3 adalah prediksi yang dihasilkan oleh model (my_model_3) terhadap data validasi (X_valid).
predictions_3 = my_model_3.predict(X_valid)

# mean_absolute_error adalah fungsi untuk menghitung Mean Absolute Error (MAE) antara prediksi (predictions_3) dan nilai sebenarnya (y_valid).
# Hasil MAE ini mengukur sejauh mana prediksi model berbeda dari nilai sebenarnya dalam skala absolut.
# Hasil MAE kemudian dicetak sebagai teks.
print("Mean Absolute Error: " + str(mean_absolute_error(predictions_3, y_valid)))


Mean Absolute Error: 0.2437787261376015


Kode di atas adalah contoh kode dalam pemrograman yang digunakan untuk menghitung Mean Absolute Error (MAE) antara prediksi model (predictions_3) dan nilai sebenarnya (y_valid) pada data validasi (X_valid). Berikut penjelasan singkat dari setiap baris kode:

1. `predictions_3 = my_model_3.predict(X_valid)`: Pada baris ini, model yang disebut `my_model_3` digunakan untuk membuat prediksi terhadap data validasi yang diberikan, yang disimpan dalam variabel `X_valid`. Hasil prediksi ini disimpan dalam variabel `predictions_3`.

2. `mean_absolute_error(predictions_3, y_valid)`: Pada baris ini, fungsi `mean_absolute_error` digunakan untuk menghitung MAE antara prediksi (`predictions_3`) dan nilai sebenarnya (`y_valid`). MAE mengukur sejauh mana prediksi model berbeda dari nilai sebenarnya dalam skala absolut.

3. `print("Mean Absolute Error: " + str(mean_absolute_error(predictions_3, y_valid)))`: Pada baris ini, hasil MAE yang telah dihitung sebelumnya dengan menggunakan fungsi `mean_absolute_error` dicetak sebagai teks. Hasil MAE ini memberikan informasi tentang sejauh mana model memiliki kesalahan dalam memprediksi data validasi. Dalam kasus ini, MAE-nya adalah sekitar 0.2438, yang menunjukkan bahwa rata-rata kesalahan prediksi model adalah sekitar 0.2438 dalam skala absolut. Semakin kecil MAE, semakin baik kinerja model dalam memprediksi data.